From 27b9e028091a2f370367e9eaf37b4bb1cde87b62 Mon Sep 17 00:00:00 2001
From: Jakub Wilk <jwilk@jwilk.net>
Date: Mon, 25 Nov 2019 07:27:12 +0100
Subject: [PATCH] =?UTF-8?q?pdf-backend:=20fix=20const=20issues=20with=20Po?=
 =?UTF-8?q?ppler=20=E2=89=A5=200.83.?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes:

    pdf-backend.cc: In static member function ‘static void pdf::Renderer::convert_path(pdf::gfx::State*, pdf::splash::Path&)’:
    pdf-backend.cc:517:40: error: invalid conversion from ‘const GfxPath*’ to ‘pdf::gfx::Path*’ {aka ‘GfxPath*’} [-fpermissive]
---
 pdf-backend.cc | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/pdf-backend.cc b/pdf-backend.cc
index 085cd9e..8a95e95 100644
--- pdf-backend.cc
+++ pdf-backend.cc
@@ -513,12 +513,11 @@ bool pdf::get_glyph(splash::Splash *splash, splash::Font *font,
 void pdf::Renderer::convert_path(pdf::gfx::State *state, splash::Path &splash_path)
 {
   /* Source was copied from <poppler/SplashOutputDev.c>. */
-  pdf::gfx::Subpath *subpath;
-  pdf::gfx::Path *path = state->getPath();
+  auto path = state->getPath();
   int n_subpaths = path->getNumSubpaths();
   for (int i = 0; i < n_subpaths; i++)
   {
-    subpath = path->getSubpath(i);
+    auto subpath = path->getSubpath(i);
     if (subpath->getNumPoints() > 0)
     {
       double x1, y1, x2, y2, x3, y3;
